### PR TITLE
ENG-16969: fix some methods to be synchronized

### PR DESCRIPTION
**Pull request by rdykiel**

### CODE REVIEW

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java [hunk 1]
```diff
@@ -808,7 +808,7 @@ public synchronized void parseAndTruncate(BinaryDequeTruncator truncator) throws
     }
 
     @Override
-    public void updateExtraHeader(M extraHeader) throws IOException {
```
Comment (luyangco) [inline src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: Why does this need to be synchronized? Is the caller always from the same thread?

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java [hunk 2]
```diff
@@ -1262,7 +1262,7 @@ public void scanEntries(BinaryDequeScanner scanner) throws IOException
     }
 
     @Override
-    public boolean deletePBDSegment(BinaryDequeValidator<M> validator) throws IOException
+    public synchronized boolean deletePBDSegment(BinaryDequeValidator<M> validator) throws IOException
```
Comment (luyangco) [inline src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: Why does this need to be synchronized? Is the caller always from the same thread?

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java (hunk 1 above)
Comment (rdykiel) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: Design consistency: the rest of the public API is synchronized. Do not make assumptions on the caller.

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java (hunk 2 above)
Comment (rdykiel) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: See previous response

**Pull request merged**